modules/data.atmosphere/R/download.ECMWF.R

```diff
@@ -0,0 +1,201 @@
+#' Downloads ECMWF Open Data 15 day forecast data products 
+#' https://confluence.ecmwf.int/display/DAC/ECMWF+open+data%3A+real-time+forecasts
+#' 
+#' https://github.com/ecmwf/ecmwf-opendata
+#' Under the hood, this function uses the Python `ecmwf-opendata` module,
+#' which can be installed via `pip` (`pip install --user ecmwf-opendata`). The
+#' module is accessed via the `reticulate` package.
+#'
+#' @param outfolder location on disk where outputs will be stored
+#' @param lat.in,lon.in site coordinates, decimal degrees (numeric)
+#' @param fchour reference time of the forecasts. Values are 00, 12 for 15 day forecast.
+#' @param stream  forecasting system that produces the data. Current value is `"enfo"`
+#' @param type the type of forecast data. Current values: `cf` (controlled forecast), `pf` (perturbed forecast)
+#' @param parameters character vector of product types, or `"all"`.
+#' @param reticulate_python Path to Python binary for `reticulate`
+#'   (passed to [reticulate::use_python()]). If `NULL` (default), use
+#'   the system default.
+#' @param overwrite Logical. If `FALSE` (default), skip any files with
+#'   the same target name (i.e. same variable) that already exist in
+#'   `outfolder`. If `TRUE`, silently overwrite existing files.
+#' @return information about the output file
+#' @export
+#' @examples
+#' 
+#' \dontrun{
+#' files <- download.ECMWF(
+#'   "ECMWF_output",
+#'   lat.in = 0,
+#'   lon.in = 180,
+#'   fchour = 00,
+#'   stream = "enfo"
+#'   type = c("cf", "pf"), 
+#'   parameters = "all", 
+#' )
+#' }
+#' @author Swarnalee Mazumder
+#'
+
+download.ECMWF <- function(outfolder, 
+                           lat.in,
+                           lon.in,
+                           fchour, 
+                           stream = "enfo", 
+                           type = c("cf", "pf"), 
+                           parameters = "all", 
+                           overwrite = FALSE, 
+                           reticulate_python = NULL,...)
+  
+{
+  if (!file.exists(outfolder)) {
+    dir.create(outfolder, showWarnings = FALSE, recursive = TRUE)
+  }
+  
+  
```

Comment:
```suggestion
  if(is.null(product)) product <- "enfo"
```

Reply:
as we decided `product` to distinguish between `open-data` and `s2s` (pending on the decision whether we want this dataset), I'm resolving this too. `download.ECMWF` function will continue to have `stream="enfo"`, `fchour=00` arguments and will in addition have `product="open-data"`